src/Components/HomeView/HomeView.js

```diff
@@ -0,0 +1,47 @@
+import './HomeView.css';
+import IntroText from './IntroText/IntroText';
+import NasdaqButton from './NasdaqButton/NasdaqButton';
+import { Link } from 'react-router-dom';
+import { fetchNasdaq, fetchNasdaqConstituents, fetchStock } from '../../apiCalls';
+import { useEffect, useState } from 'react';
+import { extractData } from '../../helperFunctions';
+
+const HomeView = () => {
+  const [nasdaqData, setNasdaqData] = useState({});
+  const [waitingForData, setWaitingForData] = useState(true);
+  const [dataFailed, setDataFailed] = useState(true);
+  const [errorMessage, setErrorMessage] = useState('')
+
+  useEffect(() => {
+    fetchNasdaq()
+      .then(data => {
+        const extractedData = extractData(data);
+        setNasdaqData(extractedData);
+      })
+      .catch((error) => {
+        setErrorMessage(error);
+        setWaitingForData(false);
+        setDataFailed(true);
+      })
+  }, [])
+
+  useEffect(() => {
+    if (nasdaqData.lastClose) {
+      setWaitingForData(false)
+      setDataFailed(false);
+    }
+  }, [nasdaqData])
```

Comment:
I think this could easily be refactored into one `useEffect` if the default value for `waitingForData`is set to true. 
You `setWaitingForData` to true in the `.then` and `.catch` and then reset it false using a `useEffect` return function.